src/main/java/seedu/contax/logic/parser/FindCommandParser.java

```diff
@@ -24,10 +30,30 @@ public FindCommand parse(String args) throws ParseException {
             throw new ParseException(
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
         }
-
-        String[] nameKeywords = trimmedArgs.split("\\s+");
-
-        return new FindCommand(new NameContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(trimmedArgs, PREFIX_SEARCH_TYPE);
+        String[] outputKeywords = args.trim().split("\\s+");
+        if (!arePrefixesPresent(argMultimap, PREFIX_SEARCH_TYPE)) {
+            return new FindCommand(new NameContainsKeywordsPredicate(Arrays.asList(outputKeywords)));
+        } else {
+            SearchType type = ParserUtil.parseSearchType(argMultimap.getValue(PREFIX_SEARCH_TYPE).get());
+            if (type.toString().equals("phone")) {
```

Comment:
Suggestion - I think using a switch case can improve the readability of this section

Reply:
thx changed alr